src/components/modals/PrintPreminedModal.tsx

```diff
@@ -0,0 +1,159 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { Form, Input, Modal, Switch } from 'antd'
+import { useForm } from 'antd/lib/form/Form'
+import { FormItems } from 'components/shared/formItems'
+import { ProjectContext } from 'contexts/projectContext'
+import { UserContext } from 'contexts/userContext'
+import { constants, Contract, utils } from 'ethers'
+import { CurrencyOption } from 'models/currency-option'
+import { useContext, useMemo, useState } from 'react'
+import { parseWad } from 'utils/formatNumber'
+
+export default function PrintPreminedModal({
+  visible,
+  onCancel,
+}: {
+  visible: boolean | undefined
+  onCancel: VoidFunction
+}) {
+  const { contracts, transactor } = useContext(UserContext)
+  const { tokenSymbol, tokenAddress, projectId, terminal } =
+    useContext(ProjectContext)
+  const [form] = useForm<{
+    beneficary: string
+    preferUnstaked: boolean
+    memo: string
+  }>()
+
+  const [currency, setCurrency] = useState<CurrencyOption>(0)
+  const [value, setValue] = useState<string>('0')
+  const [loading, setLoading] = useState<boolean>()
+
+  async function mint() {
+    if (!contracts || !projectId || !transactor || !terminal?.version) return
+
+    setLoading(true)
+
+    await form.validateFields()
+
+    let terminalContract: Contract
+    let functionName: string
+    let args: any[]
+
+    switch (terminal.version) {
+      case '1':
+        terminalContract = contracts.TerminalV1
+        functionName = 'printPreminedTickets'
+        args = [
+          projectId.toHexString(),
+          parseWad(value).toHexString(),
+          BigNumber.from(currency).toHexString(),
+          form.getFieldValue('beneficary') ?? '',
+          form.getFieldValue('memo') ?? '',
+          form.getFieldValue('preferUnstaked'),
+        ]
+        break
+      case '1.1':
+        terminalContract = contracts.TerminalV1_1
+        functionName = 'printTickets'
+        args = [
+          projectId.toHexString(),
+          parseWad(value).toHexString(),
+          form.getFieldValue('beneficary') ?? '',
+          form.getFieldValue('memo') ?? '',
+          form.getFieldValue('preferUnstaked'),
+        ]
+    }
+
+    transactor(terminalContract, functionName, args, {
+      onConfirmed: () => {
+        form.resetFields()
+        setValue('0')
+        if (onCancel) onCancel()
+      },
+      onDone: () => {
+        setLoading(false)
+      },
+    })
+  }
+
+  const projectTargetFormItemProps:
+    | { label: string; extra: string }
+    | undefined = useMemo(() => {
+    if (!terminal?.version) return
+
+    switch (terminal.version) {
+      case '1':
+        return {
+          label: 'Payment equivalent',
+          extra:
+            'The amount of tokens minted to the receiver will be calculated based on if they had paid this amount to the project in the current funding cycle.',
```

Comment:
**Question:** I'm a bit confused about this sentence. I may be understanding it wrong, but would it be better worded as:
'The amount of tokens minted to the receiver will be calculated based on the amount they paid to the project in the current funding cycle.' ?

Reply:
Lol it's a hard thing to word well. This is just terminalV1 and is more intuitive with v1.1.

the idea is to put in an amount that represents a hypothetical payment to the project. if you enter "5 ETH", the receiver will earn as many tokens as they would if they had just paid the project 5 ETH